core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -598,6 +598,11 @@ object SparkSubmit extends CommandLineUtils with Logging {
       }
     }
 
+    // In case of shells, spark.ui.showConsoleProgress can be true by default or by user.
+    if (isShell(args.primaryResource) && !sparkConf.contains(UI_SHOW_CONSOLE_PROGRESS.key)) {
```

Comment:
`!sparkConf.contains(UI_SHOW_CONSOLE_PROGRESS)`

Reply:
And, this?
```
if (isShell(args.primaryResource) && !sparkConf.contains(UI_SHOW_CONSOLE_PROGRESS)) {
...
}
```